src/systems/inter_mesh_projection.C

```diff
@@ -38,27 +49,144 @@ namespace libMesh
 
     void InterMeshProjection::project_system_vectors()
     {
-        // Number of vectors to be projected
-        unsigned int n_vecs = from_system.n_vectors();
-        libmesh_assert_equal_to (to_system.n_vectors(), n_vecs);
+        libmesh_assert_equal_to (to_system.n_vectors(), from_system.n_vectors() );
 
         // Number of variable components in each vector
         unsigned int n_vars = from_system.n_vars();
         libmesh_assert_equal_to (to_system.n_vars(), n_vars);
 
+        // We are going to use the multi-variable MeshFunction, so we can pass
+        // a single vector of variables rather than have a MeshFunction for each variable
+	    std::vector<unsigned int> variables_vector;
+
+        for (unsigned int j = 0; j != n_vars; ++j)
+        {
+          variables_vector.push_back(j);
```

Comment:
Let's add an assertion here that the names match for each variable?  It's unlikely, but I could imagine someone adding variables to the to-system in a different order than they were added to the from-system and then getting burned.

Reply:
Fixed in an upcoming commit.